exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -283,7 +290,15 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 		conditions.WithFallbackValue(ready, clusterv1.WaitingForInfrastructureFallbackReason, clusterv1.ConditionSeverityInfo, ""),
 	)
 
-	if err := r.reconcileMachines(ctx, mp, infraConfig); err != nil {
+	clusterClient, err := r.Tracker.GetClient(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Get the nodeRefsMap from the cluster.
+	s.nodeRefMapResult = r.getNodeRefMap(ctx, clusterClient)
```

Comment:
I think we should handle the error already here and then we can move it out of the getNodeRefMapResult struct (and probably don't need the struct anymore)